dispatcher/config.py

```diff
@@ -0,0 +1,77 @@
+import os
+from contextlib import contextmanager
+from typing import Optional
+
+import yaml
+
+from dispatcher.publish import DispatcherDecorator
+from dispatcher.registry import DispatcherMethodRegistry, registry
+from dispatcher.utils import resolve_callable
+
+
+class DispatcherSettings:
+    def __init__(self, config: dict, task_registry: DispatcherMethodRegistry = registry) -> None:
+        self.brokers: list = config.get('brokers', [])
+        self.producers: list = config.get('producers', [])
+        self.service: dict = config.get('service', {'max_workers': 3})
+        self.callbacks: dict = config.get('callbacks', {})
+        self.tasks: dict = config.get('tasks', {})
+        self.settings: dict = config.get('settings', {})
+
+        for task_name, options in self.tasks.items():
+            method = resolve_callable(task_name)
+            if method:
+                DispatcherDecorator(task_registry, **options)(method)
+            else:
+                raise RuntimeError(f'Could not locate task given in config: {task_name}')
+
+
+def settings_from_file(path: str) -> DispatcherSettings:
```

Comment:
should we handle IO and yaml parsing errors here?

Reply:
Is it okay if I want those to be left as exceptions? I worry that's going to be distracting work with no value-add.

Assuming a valid file, and valid YAML, we've got plenty to worry about regarding valid vs. invalid content.